### ISSUE TITLE
why large influence is harmful

### ISSUE THREAD

**Issue by zwxu064** [nimarb/pytorch_influence_functions#11]
harmful = np.argsort(influences)
Why the highest influence is harmful? note influence has positive and negative values.

**Comment by nimarb** [comment]
There is no absolute guidance on what influence numbers are harmful / helpful to the model to make a single prediction. Rather, those are in relation to one another meaning you're able to rank them.

**Comment by QingXuTHU** [comment]
I have the same question. I'm not sure whether in implementation you calculate I_{up,loss} or -1/n I_{up,loss} in original paper.

**Comment by chengrunyang** [comment]
I have the same doubts as @QingXuTHU . It seems from https://github.com/nimarb/pytorch_influence_functions/blob/66c9a9e250d663a7fc36d67ccc4cfa589bccc846/pytorch_influence_functions/calc_influence_function.py#L334 that the implementation is calculating 1/n * I_{up, loss} (z, z_test), and then take the largest as `helpful` and smallest as `harmful`. However, I think it should be the opposite: the points with negative I_{up, loss} (z, z_test) should be helpful, and those positive I_{up, loss} (z, z_test) should be harmful. I wonder if someone could help check it, thanks!

**Comment by lange-martin** [comment]
>  the points with negative I_{up, loss} (z, z_test) should be helpful, and those positive I_{up, loss} (z, z_test) should be harmful (refer to the title of Page 3, Figure 1 of the Koh and Liang paper)

I agree @chengrunyang. On page 6 of the paper, the authors also clarify that the most helpful images are those with the "most positive -I_{up, loss}" (or equivalently the most negative I_{up, loss}). Since `np.argsort` sorts the indices in ascending order, the `harmful` and `helpful` arrays should be switched in this code snippet:

https://github.com/nimarb/pytorch_influence_functions/blob/66c9a9e250d663a7fc36d67ccc4cfa589bccc846/pytorch_influence_functions/calc_influence_function.py#L346-L347

I stumbled upon this issue while testing how often a training sample is its own most influential sample. Most of the time, the training sample came out as its own most harmful sample, which is counter-intuitive. This bug in the code explains that behavior.
